### PR TITLE
Add debug-context config

### PR DESCRIPTION
**Pull request by eliblight**
A configuration option with zero impact on server operation but is printed out on server crash and can be accessed by gdb for debugging. It can be used by the user/operator to store any free-form string. This string will persist as long as the server is running and will be accessible in the following ways:

Normal config API:
```
$ src/valkey-server --debug-context test
$ src/valkey-cli config get debug-context
1) "debug-context"
2) "test"
$ src/valkey-cli config set debug-context test2
OK
```

Via GDB:
```
$ gdb -batch -ex "file src/valkey-server" -ex "attach $(pgrep valkey-server)" -ex "print server.debug_context" -ex "detach" -ex "quit" | grep " = " | awk -F\" "{print \$2}"
test2
```

And printed in crash reports:
```
------ CONFIG DEBUG OUTPUT ------
lazyfree-lazy-eviction no
...
io-threads-do-reads yes
debug-context "test2"
proto-max-bulk-len 512mb
```

### CODE REVIEW

File: src/server.h [hunk 1]
```diff
@@ -2219,6 +2219,7 @@ struct valkeyServer {
     sds availability_zone; /* When run in a cloud environment we can configure the availability zone it is running in */
     /* Local environment */
     char *locale_collate;
+    char *debug_context; /* A free-form string that can be used by the user for any purpose. It has no impact on server behavior. */
```
Comment (madolson) [inline src/server.h]:
```suggestion
    /* A free-form string that will get printed out after a crash. It is guaranteed to never impact server behavior. */
    char *debug_context; 
```
Clang format will complain about this as well.

File: src/config.c [hunk 2]
```diff
@@ -3135,6 +3135,7 @@ standardConfig static_configs[] = {
     createStringConfig("req-res-logfile", NULL, IMMUTABLE_CONFIG | HIDDEN_CONFIG, EMPTY_STRING_IS_NULL, server.req_res_logfile, NULL, NULL, NULL),
 #endif
     createStringConfig("locale-collate", NULL, MODIFIABLE_CONFIG, ALLOW_EMPTY_STRING, server.locale_collate, "", NULL, updateLocaleCollate),
+    createStringConfig("debug-context", NULL, MODIFIABLE_CONFIG | DEBUG_CONFIG, ALLOW_EMPTY_STRING, server.debug_context, "", NULL, NULL),
```
Comment (madolson) [inline src/config.c]:
```suggestion
    createStringConfig("debug-context", NULL, HIDDEN_CONFIG | MODIFIABLE_CONFIG | DEBUG_CONFIG, ALLOW_EMPTY_STRING, server.debug_context, "", NULL, NULL),
```

**Comment by madolson** [comment]
@eliblight Can you add a little bit of context about the use case to the description as well. Otherwise this seems kind of useless :).

**Comment by eliblight** [comment]
> @eliblight Can you add a little bit of context about the use case to the description as well. Otherwise this seems kind of useless :).

It just give the user a safe place to hide a string, for a (real) example, we have tests that leave orphen valkey-servers alive after the test is done. By hiding the test name in such config and later printing it by GDB attaching to valkey-server process that was left behind, we were able to pin-point the offending tests.

File: src/server.h [hunk 3]
```diff
@@ -2219,6 +2219,7 @@ struct valkeyServer {
     sds availability_zone; /* When run in a cloud environment we can configure the availability zone it is running in */
     /* Local environment */
     char *locale_collate;
+    char *debug_context; /* A free-form string that can be used for any purpose. It has no impact on server behavior. */
```
Comment (madolson) [inline src/server.h]: I would still like to include that this is printed out on server crash. The documentation should make it clear what it's intended use is. "Saying for any purpose" makes it hard to understand.

**Review by madolson — APPROVED** [review]

**Pull request merged**